examples/ai-core/src/stream-text/openai-generatetext-timeout.ts

```diff
@@ -0,0 +1,23 @@
+import { openai } from '@ai-sdk/openai';
+import { generateText } from 'ai';
+import 'dotenv/config';
+
+async function main() {
+  try {
+    const result = await generateText({
+      model: openai('gpt-3.5-turbo'),
+      prompt: 'Write a short story about a robot learning to love:\n\n',
+      timeout: 100, // Change this value to see different results
```

Comment:
can you use a custom fetch function with a delay here so timeouts can be simulated easily? ( see other examples for custom fetch )